buildSrc/src/main/java/com/google/firebase/gradle/plugins/PublishingPlugin.kt

```diff
@@ -218,18 +218,34 @@ abstract class PublishingPlugin : Plugin<Project> {
 
     if (projectsToPublish == null) return null
 
-    val projectNames = projectsToPublish.split(",")
+    val publishingProjects =
+      allFirebaseLibraries.filter { it.artifactId.get() in projectsToPublish.split(",") }
 
+    val librariesToRelease = getDeepTransitiveReleases(publishingProjects, libraryGroups)
+
+    return ReleaseMetadata(librariesToRelease, releaseName)
+  }
+
+  /**
+   * Finds all transitive libraries that need to be released so that the input list of project names
+   * can be released. This includes all project level dependencies, and anything in the same library
+   * group.
+   */
+  private tailrec fun getDeepTransitiveReleases(
```

Comment:
For this function to be tailrecursive, wouldn't it need to have as its last argument `inputGroups`? Not sure since kotlin's refdocs have a single arg function as example.

Reply:
wdym `inputGroups`?

Reply:
I meant `inputProjects` :-D 

Talked about it with @davidmotson offline, LGTM.